cpp/src/fil/infer.cu

```diff
@@ -23,6 +23,8 @@
 #include <fil/internal.cuh>
 #include "common.cuh"
 
+#define NOINLINE_FAST_COMPILE __noinline__
```

Comment:
This should default to `__forceinline__`, with a comment suggesting to change to `__noinline__` to speed up compilation during development.

Reply:
BTW, have you seen the table in the PR comments? I believe I've proven the perf impact to be negligible so that we can keep `__noinline__` as the default

Reply:
I've seen the table. I'm still unsure about changing it to `__noinline__` by default. Let's leave it at `__forceinline__` for now.